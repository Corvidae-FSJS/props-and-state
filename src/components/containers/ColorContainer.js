import React, { Component } from 'react';
import ColorDisplay from '../colors/ColorDisplay';
import ColorPicker from '../colors/ColorPicker';

export default class ColorContainer extends Component {
  state = {
    backgroundColor: 'black' //RandomColor
  };
  
  selectColor = backgroundColor => {
    this.setState({ backgroundColor });
  }

  render() {
    const { backgroundColor } = this.state;
    const colors = ['white', 'red', 'orange', 'yellow', 'green', 'blue', 'indigo', 'violet', 'black'];

    return (
      <>
        <ColorDisplay backgroundColor={backgroundColor} />
        <ColorPicker
          selectColor={this.selectColor}
          colors={colors} />
      </>
    );
  }
}