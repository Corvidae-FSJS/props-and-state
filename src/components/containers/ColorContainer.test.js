import React from 'react';
import { shallow, mount } from 'enzyme';
import ColorContainer from './ColorContainer';

describe('ColorContainer component', () => {
  it('Renders ColorContainer', () => {
    const wrapper = shallow(<ColorContainer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Changes div color on button click', () => {
    const wrapper = mount(<ColorContainer colors={[
      'red',
      'orange',
      'yellow',
      'green',
      'blue',
      'indigo', 
      'violet']} />);
    wrapper.findWhere(el => el.key() === 'black').simulate('click');
    expect(wrapper.state('backgroundColor')).toBe('black');
  });
});