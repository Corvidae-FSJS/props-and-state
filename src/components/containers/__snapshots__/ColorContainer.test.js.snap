// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorContainer component Renders ColorContainer 1`] = `
<Fragment>
  <ColorDisplay
    backgroundColor="black"
  />
  <ColorPicker
    colors={
      Array [
        "white",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "indigo",
        "violet",
        "black",
      ]
    }
    selectColor={[Function]}
  />
</Fragment>
`;
